src/BuiltInTools/dotnet-watch/HotReload/HotReloadDotNetWatcher.cs

```diff
@@ -328,15 +327,19 @@ void FileChangedCallback(ChangedPath change)
                                 fileWatcher.SuppressEvents = true;
                                 try
                                 {
-                                    var buildResults = await Task.WhenAll(
-                                        projectsToRebuild.Select(projectPath => BuildProjectAsync(projectPath, rootProjectOptions.BuildArguments, iterationCancellationToken)));
-
-                                    foreach (var (success, output, projectPath) in buildResults)
+                                    // Build projects sequentially to avoid failed attempts to overwrite dependent project outputs.
+                                    // TODO: Ideally, dotnet build would be able to build multiple projects. https://github.com/dotnet/sdk/issues/51311
+                                    var success = true;
+                                    foreach (var projectPath in projectsToRebuild)
```

Comment:
I realize that this makes everything more deterministic and easier to understand from the output, but does this result in a performance regression?

Reply:
Potentially, if there is a lot of projects that need to be rebuilt. Correctness first though. Will follow up on improving this.